controllers/terminal_controller.go

```diff
@@ -1233,6 +1233,7 @@ func (r *TerminalReconciler) createOrUpdateTerminalPod(ctx context.Context, cs *
 				pod.Spec.Tolerations = []corev1.Toleration{}
 			}
 			tolerationKey := "node-role.kubernetes.io/master"
+			criticalAddonsKey := "CriticalAddonsOnly"
```

Comment:
as we introduce another toleration key, rename `tolerationKey` to something like `masterNodeKey`